plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -111,7 +111,12 @@ func (ssl *streamSocketListener) read(c net.Conn) {
 	defer ssl.removeConnection(c)
 	defer c.Close()
 
-	scnr := bufio.NewScanner(c)
+	decoder, err := internal.NewStreamContentDecoder(ssl.ContentEncoding, c)
+	if err != nil {
+		ssl.Log.Error("Read error: %v", err)
+	}
+
+	scnr := bufio.NewScanner(decoder)
```

Comment:
We always wrap with decoder here.  Wrapping makes sense to me for gzip, but not for identity.  If identity, why not pass in c and not wrap at all?

Reply:
In the 'identity' case, internal.NewStreamContentDecoder is just returning the reader unwrapped.  I thought it would be a bit nicer to push the switch logic into this factory function.